From 9141cac175caaf176377d088e334d0991482fd6a Mon Sep 17 00:00:00 2001
From: Aaron Liu <aaronjli@google.com>
Date: Thu, 10 Aug 2023 15:38:08 +0000
Subject: [PATCH] Revert "Dismiss keyguard when simpin auth'd and..."

Revert submission 22621774-cherrypicker-L22000000959901080:N28400001357657640

Reason for revert: causing a partner bug
Fixes: 295205456
Bug: 222446076

Reverted changes: /q/submissionid:22621774-cherrypicker-L22000000959901080:N28400001357657640
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:0afa6707fa632c1f5fba5dff46a94b5d734e2ab2)
Merged-In: Icb27b4d897696b4fbb4e4a878751d925f5205dfd
Change-Id: Icb27b4d897696b4fbb4e4a878751d925f5205dfd
---
 .../android/keyguard/KeyguardSecurityContainerController.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardSecurityContainerController.java b/packages/SystemUI/src/com/android/keyguard/KeyguardSecurityContainerController.java
index 16299c7aff7b..061bab8a7006 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardSecurityContainerController.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardSecurityContainerController.java
@@ -752,7 +752,7 @@ public boolean showNextSecurityScreenOrFinish(boolean authenticated, int targetU
                 case SimPuk:
                     // Shortcut for SIM PIN/PUK to go to directly to user's security screen or home
                     SecurityMode securityMode = mSecurityModel.getSecurityMode(targetUserId);
-                    if (securityMode == SecurityMode.None || mLockPatternUtils.isLockScreenDisabled(
+                    if (securityMode == SecurityMode.None && mLockPatternUtils.isLockScreenDisabled(
                             KeyguardUpdateMonitor.getCurrentUser())) {
                         finish = true;
                         eventSubtype = BOUNCER_DISMISS_SIM;
